yagm.go

```diff
@@ -94,17 +98,31 @@ func (mux *YagmMux) HandleFunc(pattern string, handler func(http.ResponseWriter,
 func (mux *YagmMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var handler http.Handler
 
-	for _, route := range mux.routes {
-		if route.match(r.URL.Path) {
-			handler = route.handler
+    hash := sha1.New()
+    hash.Write([]byte(r.URL.Path))
 
-			requests[r] = &yagmRequest{
-				route: route,
-			}
+    if routeString, ok := mux.cache[hash]; ok {
+        var route = mux.routes[routeString]
+        handler = route.handler
 
-			break
-		}
-	}
+        requests[r] = &yagmRequest {
+            route: route,
+        }
+    } else {
+        for _, route := range mux.routes {
+            if route.match(r.URL.Path) {
+                handler = route.handler
+
+                requests[r] = &yagmRequest{
+                    route: route,
+                }
+
+                mux.cache[hash] = r.URL.Path
```

Comment:
This ain't gonna work. You need to cache the route not the requested path. It will fail in line 105.